zip-0302.rst

```diff
@@ -0,0 +1,114 @@
+::
+
+  ZIP: 302
+  Title: Standardized Memo Field Format
+  Owner: Jack Grigg <jack@electriccoin.co>
+  Original-Author: Jay Graber
+  Status: Draft
+  Category: Standards
+  Created: 2017-02-08
+  License: MIT
+
+
+Abstract
+========
+
+This ZIP describes a proposed specification for a standardized format for clients who wish
+to transmit or receive content within the encrypted memo field of shielded transactions.
+
+Motivation
+==========
+
+A well-defined standard for formatting content within the encrypted memo field will help
+expand its use within the Zcash ecosystem by providing a commonly recognized format for  
+memo values carrying different types of data. Users and third-party services benefit from 
+standardized formatting rules that define the type and length of the data contained within.
+
+Specification
+===============
+
+Section 5.5 of the Zcash protocol specification [#protocol]_ defines three cases
+for the encoding of a memo field:
+
+* a UTF-8 human-readable string [#UTF-8]_, padded by appending zero bytes; or
+* the byte ``0xF6`` followed by 511 ``0x00`` bytes, indicating "no memo"; or
+* any other sequence of 512 bytes starting with a byte value ``0xF5`` or greater (which is
+  therefore not a valid UTF-8 string), as specified in ZIP 302.
+
+This ZIP refines the specification of the third case.
+
+The following specification constrains a party, called the "reader", that interprets the
+contents of a memo. It does not define consensus requirements.
+
++ If the first byte (byte 0) has a value of 0xF4 or smaller, then the reader MUST:
+     + strip any trailing zero bytes
+     + decode it as a UTF-8 string (if decoding fails then report an error).
+
++ If the first byte has a value of 0xF5, then the reader MUST:
+     + Interpret the next few bytes (1 to 9 of them) as a 64-bit unsigned variable-length
+       integer [#Bitcoin-CompactSize]_, and use it as an arbitrary application-defined
+       "type" field.
+     + Interpret the next bytes (1 to 2 of them) as a 16-bit unsigned ULEB, and use it as
+       the length field. (The length can be at most 510 bytes due to the overall memo
+       length, and that is why the length field can only be 1 or 2 bytes.)
```

Comment:
How does an implementer decide whether to take 1 or 2 bytes when parsing? Does that depend upon the value of the `type` field?